tests/requests/efs/file_system_tests.rb

```diff
@@ -0,0 +1,146 @@
+      default_security_group = Fog::Compute[:aws].security_groups.detect { |sg| sg.description == 'default VPC security group' }
+    end
+
+    security_group = Fog::Compute[:aws].security_groups.create(
+      :vpc_id      => vpc.id,
+      :name        => "fog#{suffix}",
+      :description => "fog#{suffix}"
+    )
+
+    raises(Fog::AWS::EFS::InvalidSubnet, "invalid subnet ID: foobar#{suffix}") do
+      Fog::AWS[:efs].create_mount_target(file_system_id, "foobar#{suffix}")
+    end
+
+    raises(Fog::AWS::EFS::NotFound, "invalid file system ID: foobar#{suffix}") do
+      Fog::AWS[:efs].create_mount_target("foobar#{suffix}", subnet.identity)
+    end
+
+    if Fog.mocking?
+      tests("#create_mount_target") do
+        Fog::AWS[:efs].data[:file_systems][file_system_id]["LifeCycleState"] = 'creating'
+        raises(Fog::AWS::EFS::IncorrectFileSystemLifeCycleState) do
+          Fog::AWS[:efs].create_mount_target(file_system_id, subnet.identity)
+        end
+
+        Fog::AWS[:efs].data[:file_systems][file_system_id]["LifeCycleState"] = 'available'
+      end
+    end
+
+    raises(Fog::AWS::EFS::NotFound, "invalid security group ID: foobar#{suffix}") do
+      Fog::AWS[:efs].create_mount_target(file_system_id, subnet.identity, 'SecurityGroups' => ["foobar#{suffix}"])
+    end
+
+    tests("#create_mount_target(#{file_system_id}, #{subnet.identity})").formats(AWS::EFS::Formats::MOUNT_TARGET_FORMAT) do
+      Fog::AWS[:efs].create_mount_target(file_system_id, subnet.identity).body
+    end
+
+    tests("#describe_mount_targets(file_system_id: #{file_system_id})").formats(AWS::EFS::Formats::DESCRIBE_MOUNT_TARGETS_RESULT) do
+      Fog::AWS[:efs].describe_mount_targets(:file_system_id => file_system_id).body
+    end
+
+    mount_target_id = Fog::AWS[:efs].describe_mount_targets(:file_system_id => file_system_id).body["MountTargets"].first["MountTargetId"]
+
+    tests("#describe_mount_target_security_groups(#{mount_target_id})").formats(AWS::EFS::Formats::DESCRIBE_MOUNT_TARGET_SECURITY_GROUPS_FORMAT) do
+      result = Fog::AWS[:efs].describe_mount_target_security_groups(mount_target_id).body
+      returns([default_security_group.group_id]) { result["SecurityGroups"] }
+      result
+    end
+
+    raises(Fog::AWS::EFS::Error, 'Must provide at least one security group.') do
+      Fog::AWS[:efs].modify_mount_target_security_groups(mount_target_id, [])
+    end
+
+    tests("#modify_mount_target_security_groups(#{mount_target_id}, [#{security_group.group_id}])") do
+      returns(204) do
+        Fog::AWS[:efs].modify_mount_target_security_groups(mount_target_id, [security_group.group_id]).status
+      end
+    end
+
+    tests("#describe_mount_target_security_groups(#{mount_target_id})").formats(AWS::EFS::Formats::DESCRIBE_MOUNT_TARGET_SECURITY_GROUPS_FORMAT) do
+      result = Fog::AWS[:efs].describe_mount_target_security_groups(mount_target_id).body
+      returns([security_group.group_id]) { result["SecurityGroups"] }
```

Comment:
It is definitely eventual consistency.  I don't like the idea of a wait_for block here though, since there is nothing to definitively wait for.  The state of the mount target doesn't change when you modify the security groups.  Nothing can really be done except repeated queries, which seems hacky.